samples/web/Continote/public/scripts/auth-helper.js

```diff
@@ -83,14 +85,104 @@ function AuthHelper(handleUserSignedIn, handleUserSignedOut) {
   };
 
   /**
-   * FirebaseUI will be initialized within the element on the page with this ID.
+   * FirebaseUI will be initialized within the element on the page with this id.
+   *
+   * That element is added with the signInUITemplate_ below.
    *
    * @type {!string}
    * @const
    */
   var firebaseUiSignInContainerId_ = "firebaseui-container";
 
   /**
+   * A sign in UI will be initialized within the element on the page with this
+   * id.
+   *
+   * @type {!string}
+   * @const
+   */
+  var signInUiContainerId_ = "sign-in-container";
+
+  /**
+   * Element template used to add a sign in UI to the page.
+   *
+   * @type {!string}
+   * @const
+   */
+  var signInUITemplate_ =
+      '<section class="mdl-grid mdl-grid--no-spacing mdl-shadow--2dp">' +
+        '<div class="mdl-card mdl-cell mdl-cell--12-col">' +
+          '<div class="mdl-card__title">' +
+            '<h2 class="mdl-card__title-text">' +
+            'Sign in required' +
+            '</h2>' +
+          '</div>' +
+          '<div class="mdl-card__supporting-text">' +
+            '<p>Please sign in below to view this page.</p>' +
+            '<div id="'+ firebaseUiSignInContainerId_ +'"></div>' +
```

Comment:
I made it a template to re-use the "appendElementBasedOnTemplateToContainer" utility function, which is used to add Notes to the Note list based on a template.